core/storage/pager.rs

```diff
@@ -499,27 +498,38 @@ pub enum BtreePageAllocMode {
 struct CommitInfo {
     completions: Vec<Completion>,
     state: CommitState,
-    /// Dirty page IDs to process during PrepareFrames (for async evicted page reads)
-    dirty_ids_to_process: Vec<usize>,
-    /// Current index into dirty_ids_to_process
-    current_dirty_idx: usize,
-    /// Pages collected so far during PrepareFrames
     collected_pages: Vec<PageRef>,
-    /// Pending read for an evicted page: (page_id, page_ref, read_completion)
-    pending_read: Option<(usize, PageRef, Completion)>,
+    page_sources: Vec<PageSource>,
+    page_source_cursor: usize,
     prepared_frames: Vec<PreparedFrames>,
 }
 
+enum PageSource {
+    // page_id fetched from cache during PrepareFrames
+    Cached(usize),
+    // pre-read page
```

Comment:
this comment is weird. "pre-read" sounds like it's cached, but it says `Evicted`. Also the comments should use triple slashes `///` :]